src/Picqer/Financials/Exact/StockPosition.php

```diff
@@ -28,12 +28,27 @@ class StockPosition extends Model
      *
      * @var string
      */
-    protected $url = 'logistics/StockPosition';
+    protected $url = 'read/logistics/StockPosition';
 
     /**
      * The primary key for the current entity.
      *
      * @var string
      */
     protected $primaryKey = 'ItemId';
+
+    /**
+     * Find a stock position with item id
+     *
+     * @param string $id For example: 00000000-0000-0000-0000-000000000000
+     * @return StockPosition
+     */
+    public function find($id)
+    {
+        $result = $this->connection()->get($this->url, [
+            'itemId' => sprintf('guid\'%s\'', $id)
```

Comment:
Doesn't the need to override this method simply mean that the `$pimaryKey` property isn't set correctly? Could you test what happens when you update that property?